lib/xmlhttprequest.js

```diff
@@ -112,23 +99,53 @@ function _receiveResponse(response) {
   _properties.responseText = '';
   _readyStateChange.call(this, XMLHttpRequest.LOADING);
   const textResponseType = !['arraybuffer', 'blob'].includes(_properties.responseType);
-  if (textResponseType) {
-    response.setEncoding('utf8');
-  }
-  response.addListener('data', chunk => {
-    if (!textResponseType) {
-      _properties.responseBuffers.push(chunk);
-    } else {
-      _properties.responseText = _properties.responseText + chunk;
-    }
-    this.loaded += chunk.length;
+
+  _readyStateChange.call(this, XMLHttpRequest.HEADERS_RECEIVED);
+
+  const isStream = (typeof response.body.pipe) === 'function';
```

Comment:
Is there a better way to detect this with node's modules? And will there always be a `body`?

Reply:
There _should_ always be a body, but I'll check it out.